spec/models/user_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject do
+    User.new(
+      name: 'test',
+      photo: 'https://test.com/profile.png',
+      bio: 'I am test running.',
+      post_counter: 0
+    )
+  end
+
+  before { subject.save }
+
+  context 'Return valid data' do
+    it 'should accept the name' do
+      subject.name = 'test'
+      expect(subject).to be_valid
+    end
+
+    it 'should accept post_counter' do
+      subject.post_counter = 3
+      expect(subject).to be_valid
+    end
+  end
+
+  context 'Return invalid data' do
+    it 'should not accept blank name' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'should not accept negative post_counter' do
+      subject.post_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'should not accept non numerical post_counter' do
+      subject.post_counter = 'count'
+      expect(subject).to_not be_valid
+    end
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.), Please add a unit test for the `recent_three_posts` method in user model.